trtools/dumpSTR/dumpSTR.py

```diff
@@ -514,102 +524,167 @@ def GetAllCallFilters(call_filters):
+                     sample_names: List[str]) -> trh.TRRecord:
+    r"""Apply call-level filters to a record.
 
-    Returns a new record with FILTER populated for each sample.
+    Returns a TRRecord object with the FILTER (or DUMPSTR_FILTER)
+    format field updated for each sample.
     Also updates sample_info with sample level stats
 
     Parameters
     ----------
-    record : vcf._Record
-       The record to apply filters to
-    reader : vcf.Reader
-       The vcf.Reader object we're reading from
-    call_filters : list of filters.Reason
+    record :
+       The record to apply filters to.
+       Note: once this method has been run, this object
+       will be in an inconsistent state. All further use
+       should be directed towards the returned TRRecord object.
+    call_filters :
        List of call filters to apply
-    sample_info : dict of str->value
-       Dictionary of sample stats to keep updated
+    sample_info :
+       Dictionary of sample stats to keep updated,
+       from name of filter to array of length nsamples
+       which counts the number of times that filter has been
+       applied to each sample across all loci
+    sample_names:
+        Names of all the samples in the vcf. Used for formatting
+        error messages.
 
     Returns
     -------
-    new_record : vcf._Record
-       Modified record object with FILTER field set
+    trh.TRRecord
+        A reference to the same underlying cyvcf2.Variant object,
+        which has now been modified to contain all the new call-level
+        filters.
     """
-    # Add FILTER to end of formats
-    if "FILTER" in record.FORMAT:
-        samp_fmt = vcf.model.make_calldata_tuple(record.FORMAT.split(':'))
+    all_filter_text = np.empty((record.GetNumSamples()), 'U4')
+    nocalls = ~record.GetCalledSamples()
+
+    for filt in call_filters:
+        filt_output = filt(record)
+        # This will throw a TypeError if passed a non numeric
+        # array. Will need better logic here if we decide to create
+        # call level filters which return nonnumeric output
+        nans = np.isnan(filt_output)
+        if np.all(nans):
+            continue
+        sample_info[filt.name] += np.logical_and(~nans, ~nocalls)
+        filt_output_text = np.char.mod('%g', filt_output)
+        filt_output_text = np.char.add(filt.name, filt_output_text)
+        filt_output_text[nans] = ''
+        not_first_filter = np.logical_and(~nans, all_filter_text != '')
+        all_filter_text[not_first_filter] = \
+            np.char.add(all_filter_text[not_first_filter], ',')
+        all_filter_text = np.char.add(all_filter_text, filt_output_text)
+
+    # append NOCALL to each sample that has not been called
+    nocalls = ~record.GetCalledSamples()
+    if np.any(nocalls):
+        nocall_text = np.empty((nocalls.shape[0]), dtype='U6')
+        nocall_text[nocalls] = 'NOCALL'
+        # if there was already a no call, don't say we applied
+        # our filters to it
```

Comment:
Just to make clear what currently happens: Take this example as input to DumpSTR
GT:Q     .|.:5
Since it's already filtered, I had DumpSTR emit
GT:Q:FILTER     .|.:5:
and not
GT:Q:FILTER     .|.:.:NOCALL

That way it's clear what DumpSTR filtered and what it didn't. But I'm not sold on that choice. Do you prefer the alternative?

Reply:
humm I think the Filter reason is missing on the example that you gave (for what you currently emit). I think if the FILTER field is incluced, it should be better than the last one.

Reply:
Sorry, my understanding/explanation above was just wrong. If a field was already filtered, then we append "NOCALL" to it but don't append any other filter reasons. E.g. if we have --min-q 20,
GT:Q .|.:5
becomes
GT:Q:FILTER .|.:5:NOCALL
and not
GT:Q:FILTER .|.:5:MINQ20_5,NOCALL